esmvaltool/preprocessor/_derive/__init__.py

```diff
@@ -99,15 +96,19 @@ def derive(cubes,
     if fx_files:
         for (fx_var, fx_path) in fx_files.items():
             if fx_path is not None:
-                cubes.append(iris.load_cube(fx_path))
+                fx_cube = iris.load_cube(
+                    fx_path,
+                    constraint=iris.Constraint(
+                        cube_func=lambda c, var=fx_var: c.var_name == var))
+                cubes.append(fx_cube)
```

Comment:
what happens if it fails to load the cube eg if the fx variable data/metadata is crap )had the case for a couple CMIP5 files and in that case we don't apply the fx mask but use Natural Earth instead)  or the cube is in fact a `cubeList`?

Reply:
Good question...can we apply fixes to the fx files? For some CMIP6 data I need this constraint, if there are CMIP5 files with the wrong names we should fix those files. As I use `iris.load_cube()`, the cube is always a single cube.